cmd/certsuite/upload/results_spreadsheet/results_spreadsheet.go

```diff
@@ -0,0 +1,367 @@
+	indices, err := getHeaderIndicesByColumnNames(headers, []string{"CNFName"})
+	if err != nil {
+		return nil, err
+	}
+	workloadIndex := indices[0]
+
+	// add to sheet only rows of given workload name
+	for _, row := range rawResultsSheet.Data[0].RowData[1:] {
+		if len(row.Values) > workloadIndex {
+			curWorkloadName := *row.Values[workloadIndex].UserEnteredValue.StringValue
+			if curWorkloadName == workloadName {
+				// add empty values in 2 added columns
+				newRow := &sheets.RowData{
+					Values: append([]*sheets.CellData{{}, {}}, row.Values...),
+				}
+				filteredRows = append(filteredRows, newRow)
+			}
+		}
+	}
+
+	workloadResultsSheet := &sheets.Sheet{
+		Properties: &sheets.SheetProperties{
+			Title: "results",
+		},
+		Data: []*sheets.GridData{{RowData: filteredRows}},
+	}
+
+	return workloadResultsSheet, nil
+}
+
+func createSingleWorkloadRawResultsSpreadSheet(sheetService *sheets.Service, driveService *drive.Service, folder *drive.File, rawResultsSheet *sheets.Sheet, workloadName string) (*sheets.Spreadsheet, error) {
+	workloadResultsSheet, err := createSingleWorkloadRawResultsSheet(rawResultsSheet, workloadName)
+	if err != nil {
+		return nil, err
+	}
+
+	workloadResultsSpreadsheet := &sheets.Spreadsheet{
+		Properties: &sheets.SpreadsheetProperties{
+			Title: fmt.Sprintf("%s Best Practices Test Results", workloadName),
+		},
+		Sheets: []*sheets.Sheet{workloadResultsSheet},
+	}
+
+	workloadResultsSpreadsheet, err = sheetService.Spreadsheets.Create(workloadResultsSpreadsheet).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	if err := addFilterByFailedAndMandatoryToSheet(sheetService, workloadResultsSpreadsheet, "results"); err != nil {
+		return nil, err
+	}
+
+	if err := MoveSpreadSheetToFolder(driveService, folder, workloadResultsSpreadsheet); err != nil {
+		return nil, err
+	}
+
+	return workloadResultsSpreadsheet, nil
+}
+
+//nolint:funlen
+func createConclusionsSheet(sheetsService *sheets.Service, driveService *drive.Service, rawResultsSheet *sheets.Sheet) (*sheets.Sheet, error) {
+	headers := getHeadersFromSheet(rawResultsSheet)
+	colsIndices, err := getHeaderIndicesByColumnNames(headers, []string{"CNFName", "CNFType", "OperatorVersion"})
+	if err != nil {
+		return nil, err
+	}
+
+	workloadNameColIndex := colsIndices[0]
+	workloadTypeColIndex := colsIndices[1]
+	operatorVersionColIndex := colsIndices[2]
+
+	// Initialize sheet with headers
+	conclusionsSheetRowsValues := []*sheets.CellData{}
+	for _, colHeader := range conclusionSheetHeaders {
+		headerCellData := &sheets.CellData{UserEnteredValue: &sheets.ExtendedValue{StringValue: stringToPointer(colHeader)}}
+		conclusionsSheetRowsValues = append(conclusionsSheetRowsValues, headerCellData)
+	}
+	conclusionsSheetRows := []*sheets.RowData{{Values: conclusionsSheetRowsValues}}
+
+	// Extract unique values from the CNFName column and fill sheet
+	uniqueMap := make(map[string]bool)
+	for _, rawResultsSheetrow := range rawResultsSheet.Data[0].RowData[1:] {
```

Comment:
Same comment here as at line 157: `len(rawResultsSheet.Data[0].RowData)` should be checked to be `> 1`. Otherwise, it must be explicitly stated in the function documentation that the caller is responsible to check it before calling this func.

Reply:
I agree, I will add this check in the `createConclusionSheet` func, and as the `createSingleWorkloadRawResultsSheet` is called only from there, the check will sufficient for both cases.